src/bootstrap/test.rs

```diff
@@ -783,6 +783,13 @@ impl Step for RustdocJSStd {
                 .arg(builder.doc_out(self.target))
                 .arg("--test-folder")
                 .arg(builder.src.join("src/test/rustdoc-js-std"));
+            for path in &builder.paths {
+                if let Some(name) = path.file_name().and_then(|f| f.to_str()) {
+                    if name.ends_with(".js") {
```

Comment:
This is probably not quite right -- the typical filter can be found in this file, if you search for "Get test-args by striping suite path" -- the logic is somewhat more complicated. Extracting that into a helper function seems like a good idea to me.

Reply:
Indeed! I'll extract the function then.

Reply:
I can't re-use the same code because I need the full path. However, I think I 
can do something similar.

Reply:
I can't re-use the same code because I need the full path. However, I think I can do something similar.